File: client/src/api/event.js
import { API } from "./const";
import axios from "axios";

export async function getEvents() {
  const res = await fetch(API + "/event", {
    method: "GET",
  });

  return res.json();
}

export async function getEventLists(query) {
  const uuid = query.queryKey[1];
  const res = await fetch(API + `/event/${uuid}` + "/lists");

  return res.json();
}

export async function postEvent(data) {
  return axios.post(API + "/event", data);
}

export async function deleteEvent(uuid) {
  return axios.delete(API + "/event/" + uuid);
}

export async function updateEvent(event) {
  return axios.patch(API + "/event", event);
}

File: server/internal/route/event.go
package route

import (
	"github.com/gofiber/fiber/v2"
	"github.com/jmoiron/sqlx"
	"github.com/meowshi/pasco-server/internal/controller"
	"github.com/meowshi/pasco-server/internal/storage/postgres"
	internal_usecase "github.com/meowshi/pasco-server/internal/usecase"
	"github.com/meowshi/pasco-server/pkg/storage"
	"github.com/meowshi/pasco-server/pkg/usecase"
	"google.golang.org/api/sheets/v4"
)

func SetupEventRouter(app *fiber.App, db *sqlx.DB, service *sheets.Service) {
	var eventStorage storage.EventStorage = postgres.NewEventPostgresStorage(db)
	var envStorage storage.EnvStorage = postgres.NewEnvPostgresStorage(db)
	var yandexoidStorage storage.YandexoidStorage = postgres.NewYandexoidPostgresStorage(db)
	var registrationStorage storage.RegistrationStorage = postgres.NewRegistrationPostgresStorage(db)

	var usecase usecase.EventUsecase = internal_usecase.NewEventUsecase(service, eventStorage, registrationStorage, yandexoidStorage, envStorage)

	controller := controller.NewEventController(usecase)

	router := app.Group("/event")

	router.Post("", controller.Create)
	router.Get("", controller.GetByDate)
	router.Get("/:uuid", controller.Get)
	router.Get("/:uuid/lists", controller.GetEventLists)
	router.Delete("/:uuid", controller.Delete)
	router.Patch("", controller.Update)
}
